pkg/dispatcher/dispatcher.go

```diff
@@ -179,3 +189,21 @@ func isSuccess(ctx context.Context, result protocol.Result) bool {
 	logging.FromContext(ctx).Warnf("Invalid result type, not RetriesResult")
 	return false
 }
+
+func readSpan(ctx context.Context, msg wabbit.Delivery) (context.Context, *trace.Span) {
+	traceparent, ok := msg.Headers()["traceparent"].(string)
+	if !ok {
+		return ctx, nil
+	}
+	tracestate, ok := msg.Headers()["tracestate"].(string)
+	if !ok {
+		return ctx, nil
+	}
+	sc, ok := (&tracecontext.HTTPFormat{}).SpanContextFromHeaders(traceparent, tracestate)
+	var span *trace.Span
+	if ok {
+		ctx, span = trace.StartSpanWithRemoteParent(ctx, "rabbitmq_dispatcher", sc,
```

Comment:
nit/thought, if we move the const component from main into this file, then we could use it in the main and here as well and wouldn't have to worry about them going out of sync. Just jotting down, not a big deal.